### PR TITLE
Add snapshot test to others generators.

### PR DESCRIPTION
**Pull request by mshima**
Following https://github.com/jhipster/generator-jhipster/pull/14852.

This PR adds snapshot tests to others generators that don't have regression tests in CI.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
@Tcharl snapshot tests for the others generators.
It took some time but it would be hard to fix regressions in deployments generators.

File: test/__snapshots__/ci-cd.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,3754 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (Tcharl) [inline test/__snapshots__/ci-cd.spec.js.snap:1]: Shouldn't these one be ignored by git?

**Comment by Tcharl** [comment]
Awesome!
Would it be interesting to generalize snapshot testing? Looks like it will dramatically harden the jhipster tests as well as simplify their maintenance (total coverage can become a no-brainer: one file + all the combinations of options it uses == as many tests).
WDYT?

**Review by Tcharl — APPROVED** [review]

**Comment by mshima** [comment]
> Would it be interesting to generalize snapshot testing?

Some generators generates too much code, and the files would be too big.

Follow-ups:
- liquibase/changelog generator?
- replace manually created expected files: https://github.com/mshima/generator-jhipster/blob/main/test/utils/expected-files.js with filenames snapshots.
- drop old tests from deployments.

**Comment by Tcharl** [comment]
Could be nice for liquibase/incremental: I encountered some issues with some ids and relationships.
Big +1 for expected files of course: free food :-)

**Pull request merged**